### PR TITLE
Apb63 merge branch

### PR DESCRIPTION
**Pull request by howzat**
I've updated the RAML and modularised it (with more examples). So far only the agencies and clients are defined in the RAML but only agencies in XML.

### CODE REVIEW

File: agent-client-authorisation.raml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    arn:
+      description: The MTD platform Agent Registration Number
+      type: string
+  get:
+    description: List of invitations sent by the agency (arn)
+    queryParameters:
+      status:
+        description: filter invitations by status.
+        type: invitations.Status
+      customerRegimeId:
```
Comment (bravesirrobin81) [inline agent-client-authorisation.raml]: s/customer/client/g?

File: agent-client-authorisation.raml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        description: filter invitations by status.
+        type: atoms.Status
+    responses:
+      200:
+        description: A list of invitations.
+        body: invitations.Invitation[]
+      401:
+        description: The client must be authenticated and authorised (logged-in) to use this resource
+      403:
+        description: The client is not permitted to see this invitation.
```
Comment (bravesirrobin81) [inline agent-client-authorisation.raml]: "these invitations"

File: public/api/documentation/0.95a/Agency-Cancel-invitation.xml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+<endpoint>
+    <name>Agency Cancels and Invitation</name>
```
Comment (bravesirrobin81) [inline public/api/documentation/0.95a/Agency-Cancel-invitation.xml]: s/and/an/

File: public/api/documentation/0.95a/Agency-Cancel-invitation.xml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        <authorisation>
+            <type>USER</type>
+            <scope>write:sent-invitations</scope>
+        </authorisation>
+    </section>
+    <section id="content-types">
+        <title>Accepted Media Types</title>
+        <list>
+            <item>
+                <code>application/vnd.hmrc.1.0+json</code>
```
Comment (bravesirrobin81) [inline public/api/documentation/0.95a/Agency-Cancel-invitation.xml]: This doesn't match the media type in the RAML.

File: public/api/documentation/0.95a/Agency-Create-an-invitation.xml [hunk 5]
```diff
@@ -0,0 +1,144 @@
+<endpoint>
+    <name>Agency Creates and Invitation</name>
```
Comment (bravesirrobin81) [inline public/api/documentation/0.95a/Agency-Create-an-invitation.xml]: s/and/an/

File: public/api/documentation/0.95a/Agency-Sent-invitation.xml [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+			"clientRegimeId": "123456789",
+			"status": "pending"
+		}]]></json>
+        <table>
+            <rows>
+
+                <row>
+                    <cell><code>_links["self"].href</code></cell>
+                    <cell><code>URI</code></cell>
+                    <cell><code>/url/to/cancel/this/invitation</code></cell>
```
Comment (bravesirrobin81) [inline public/api/documentation/0.95a/Agency-Sent-invitation.xml]: Remove "cancel"?

File: raml/atoms.raml [hunk 7]
```diff
@@ -3,6 +3,11 @@ usage: A basic type in the agent client authorisation raml library
 
 types:
 
+  Timestamp:
+    type: datetime
+    example: Sun, 28 Feb 2016 16:41:41 GMT
+    format: rfc2616
```
Comment (frankoid) [inline raml/atoms.raml]: How come we're using RFC 2616 datetime format not ISO 8601 as per https://developer.service.hmrc.gov.uk/api-documentation/docs/reference-guide#common-data-types

Comment (bravesirrobin81) [inline raml/atoms.raml]: Probably because I picked it at random. If we omit the format, we'll get RFC3339 which is ISO 8601.

Comment (frankoid) [inline raml/atoms.raml]: I think we should do that (omit the format so we get RFC3339 == ISO8601) and update the example to match

File: agent-client-authorisation.raml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        type: string
+    delete:
+      description: Cancel an invitation.
+      responses:
+        204:
+          description: Invitation has been cancelled.
+        401:
+          description: The agent must be authenticated and authorised (logged-in) to use this resource
+        403:
+          description: The agent is not permitted to delete this invitation.
```
Comment (bravesirrobin81) [inline agent-client-authorisation.raml]: Do we want to distinguish here between 'not permitted' as in cancelling someone else's invitation, and 'not permitted' as in 'the invitation has already been accepted'?

File: raml/atoms.raml (hunk 7 above)
Comment (frankoid) [inline raml/atoms.raml]: https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14

**Pull request merged**